### PR TITLE
Improve cover image performance

### PR DESCRIPTION
**Pull request by mikiher**
Following the experiments described [here](https://github.com/advplyr/audiobookshelf/discussions/3570), I made the following changes to cover image GET requests (I will do the same for author images in separate PR - wanted to focus on covers here):
- Removed authentication and user deserialization (req.user) from cover requests
- Optimized `LibraryItemController.getCover` so that no database access is made if the resized image is already in the covers disk cache
- Removed the token parameter from cover urls on the web client

This reduces the average completion time for cover requests by a factor of ~10 when acessing the server through https, and minimizes concurrent access to the database.

### CODE REVIEW

**Comment by advplyr** [comment]
Thanks!

File: server/Auth.js [hunk 1]
```diff
@@ -18,6 +18,26 @@ class Auth {
...
+
+  /**
+   * Checks if the request should not be authenticated.
+   * @param {Request} req
+   * @returns {boolean}
+   * @private
+   */
+  authNotNeeded(req) {
+    return req.method === 'GET' && this.ignorePattern.test(req.originalUrl)
+  }
```
[the remark this answers is not in the document]
Comment (mikiher) [reply server/Auth.js:32]: Yes, it does. We discussed this and concluded that the security risk is not very high in this case.
 
The situation up until now was much worse in my opinion - we sent authentication token as a url query parameter in cover and author img tags, and now we don't do that any more.

[the remark this answers is not in the document]
Comment (devnoname120) [reply server/Auth.js:32]: @mikiher It's a lot of work but usually people solve this problem using HMAC signatures that is provided in the GET parameters of the URL.

[the remark this answers is not in the document]
Comment (mikiher) [reply server/Auth.js:32]: Yes, I read about some of these solutions a while back. It's not worth the investment, IMO. This isn't a DoD app 🙂

**Comment by watsonbox** [comment]
Excellent performance investigation and improvements, thanks a lot 👏

**Pull request merged**